File: cpp/SesiTayang.cpp
#include <iostream>
#include <string>
#include <iomanip> // Memformat output presisi angka

using namespace std;

// Representasi entitas Sesi Tayang Bioskop
class SesiTayang
{
    private:
        // Enkapsulasi atribut untuk melindungi state objek
        string id_sesi;
        string judul_film;
        string jadwal;
        double harga;

    public:
    // Constructor untuk inisialisasi awal objek
    SesiTayang(string id, string judul, string jadwal, double harga)
    {
        setId(id);
        setJudul(judul);
        setJadwal(jadwal);
        setHarga(harga);
    }

    // --- Mutator (Setter) ---
    void setId(const string& id) 
    {
        this->id_sesi = id;
    }

    void setJudul(const string& judul) 
    {
        this->judul_film = judul;
    }

    void setJadwal(const string& jadwal) 
    {
        this->jadwal = jadwal;
    }

    void setHarga(const double& harga) 
    {
        this->harga = harga;
    }

    // --- Accessor (Getter) ---
    string getId() const
    {
        return id_sesi;
    }

    string getJudul() const
    {
        return judul_film;
    }

    string getJadwal() const
    {
        return jadwal;
    }

    double getHarga() const
    {
        return harga;
    }
    
    // Menampilkan detail sesi tayang ke konsol
    void tampilkanData() const 
    {
        cout << "ID Sesi : " << getId() << endl
            << "Judul Film : " << getJudul() << endl
            << "Jadwal Tayang : " << getJadwal() << endl
            << "Harga Tiket : " << fixed << setprecision(0) << getHarga() << endl;
    }
    
    // Destructor
    ~SesiTayang()
    {
    }
};

File: java/SesiTayang.java
// Representasi entitas Sesi Tayang Bioskop
public class SesiTayang
{
    // Enkapsulasi atribut state
    private String id_sesi;
    private String judul_film;
    private String jadwal;
    private double harga;

    // Constructor untuk inisialisasi awal
    public SesiTayang(String id_sesi_baru, String judul_film_baru, String jadwal_baru, double harga_baru)
    {
        this.id_sesi = id_sesi_baru; 
        this.judul_film = judul_film_baru; 
        this.jadwal = jadwal_baru; 
        this.harga = harga_baru; 
    }

    // --- Accessor (Getter) ---
    public String getId()
    {
        return id_sesi; 
    }

    public String getJudul()
    {
        return judul_film; 
    }

    public String getJadwal()
    {
        return jadwal; 
    }

    public double getHarga()
    {
        return harga; 
    }

    // --- Mutator (Setter) ---
    public void setId(String id_sesi)
    {
        this.id_sesi = id_sesi; 
    }

    public void setJudul(String judul_film)
    {
        this.judul_film = judul_film; 
    }

    public void setJadwal(String jadwal)
    {
        this.jadwal = jadwal; 
    }

    // Setter khusus dengan validasi input harga
    public void setHarga(double harga)
    {
        if (harga > 0) 
        {
            this.harga = harga; 
        }
        else
        {
            System.out.println("Error: Harga harus lebih dari 0.");
        }
    }

    // Mencetak informasi entitas ke konsol
    void tampilkanData()
    {
        System.out.println("ID Sesi : " + getId());
        System.out.println("Judul Film : " + getJudul());
        System.out.println("Jadwal Tayang : " + getJadwal());
        System.out.printf("Harga Tiket : %.0f\n", getHarga());
    }
}